### ISSUE TITLE
Ignore unrecognized fields

### ISSUE THREAD

**Issue by darcling** [FieldVal/fieldval-js#21]
I forked you to add the option to ignore unrecognized fields because my UI framework adds some but my Node.js API doesn't need to care.  But, while reading through your code I noticed that I can get effectively the same results by instantiating my validator with a null second parameter (existing_error) : 
new FieldVal(input, null);

But, I think that's just a coincidence since it still does unrecognized logic checking in generate_response().  

I took a quick stab at accepting options without breaking backwards compatibility over here : 
https://github.com/darcling/fieldval-js/commit/76cfc059a8297baa78dc460210fe981b41ff7fb0
But I only added the 'ignore unrecognized fields' option because that's what I needed : )

You may have a better way.  

Thanks for the library by the way, I'm just getting rolling with it.

**Comment by MarcusLongmuir** [comment]
Hey @darcling. I hope the library is useful.

I looked at your change and decided the best approach was to change the constructor to accommodate ```options``` as the second (optional) argument. The ```existing_error``` argument can now be supplied using the ```"error"``` property in the options. I've also added support for your additional functionality, ```"ignore_unrecognized"```. These changes are currently on the ```develop_options``` branch (60a137b4c597e1cae8933d89a4f0ccd2ff1b566b). I'll merge them into a release soon if I find no severe issues with projects this library is being used in.

```javascript
new FieldVal(my_data,{
    "error": my_existing_error,
    "ignore_unrecognized": true
});
```
